src/lib/djerba/plugins/genomic_landscape/plugin.py

```diff
@@ -96,16 +99,36 @@ def extract(self, config):
         plugin_dir = os.path.dirname(os.path.realpath(__file__))
         r_script_dir = os.path.join(plugin_dir, 'Rscripts')
 
-        # Make a file where all the (actionable) biomarkers will go
+        # Make a file where all the (actionable) biomarkers will go, and initialize results
         biomarkers_path = self.make_biomarkers_maf(work_dir)
         results = tmb_processor(self.log_level, self.log_path).run(
             work_dir, data_dir, r_script_dir, tcga_code, biomarkers_path, tumour_id
         )
-        results[glc.PURITY] = wrapper.get_my_float(glc.PURITY_INPUT) * 100
+        # evaluate HRD and MSI reportability
+        purity = wrapper.get_my_float(glc.PURITY_INPUT)
+        sample_type = wrapper.get_my_string(glc.SAMPLE_TYPE)
+        sample_is_ffpe = False
+        if re.search('FFPE', sample_type.upper()):
+            sample_is_ffpe = True
+            self.logger.debug('FFPE sample detected')
+        elif sample_type == glc.UNKNOWN_SAMPLE_TYPE:
+            self.logger.warning("Unknown sample type in config; assuming non-FFPE sample")
+        else:
+            self.logger.debug('Non-FFPE sample detected')
+        if purity > 0.5 or (purity > 0.3 and not sample_is_ffpe):
```

Comment:
Doesn't the logic in the first if-statement encompass the logic in the second if-statement?
Also, the purities should be >= 0.5 and >= 0.3